### PR TITLE
pbench-fio: compute localhost latency profiles

### LINKED ISSUE #2061: pbench-fio latency histograms only get generated when passing --client
To understand, inter alia, public cloud storage performance under etcd load (and thus whether or not it can handle the load of being used on an Openshift controller node) we need to track the 99th percentile latency for fdatasync() calls.  In trying to simulate that we discovered that the latency data wasn't being graphed, and in a chat with @portante he found that the code is there to do it but surprisingly only runs if the benchmark is run using --client.  The data is perfectly valid for a single host, so there's no reason for that restriction.

### PR DESCRIPTION
**Pull request by portante**
Fixes #2061.

### CODE REVIEW

File: agent/bench-scripts/pbench-fio
Comment (dbutenhof) [inline agent/bench-scripts/pbench-fio:731]: There seem to be unnecessary differences from the clients code above. Specifically, why put the results into `clients/localhost/hist` instead of `hist`? Without that difference it seems that this post-processing step would be identical with or without `clients` and could just be factored out of the `if/else`.

File: agent/bench-scripts/tests/pbench-fio/test-04.txt
Comment (dbutenhof) [inline agent/bench-scripts/tests/pbench-fio/test-04.txt:27]: So awkward to put together all the pieces of the test command structure to figure out whether these gold file changes are good. I'm assuming this must indicate a single-host fio where we were previously not looking for these logs; but there seem to be rather a lot of them. And are there (single host) test cases were we do have the logs to validate that they're processed now?

File: agent/bench-scripts/pbench-fio
Comment (portante) [reply agent/bench-scripts/pbench-fio:731]: Yeah, it's ugly.  I'll attempt a refactor to unify this behavior.

File: agent/bench-scripts/tests/pbench-fio/test-04.txt
Comment (portante) [reply agent/bench-scripts/tests/pbench-fio/test-04.txt:27]: This was an attempt to make a minimal change without opening a can of worms.  Unfortunately, it failed.

By making this change, almost all of our fio tests have latency profiles that were not being computed, and so we have this deluge of changes in the test environment.

**Comment by frival** [comment]
From our earlier conversation:
+1 The only thing that makes me a little sad is that it goes from 99 to 100 without an intervening step.  In RGW land we found 99.5 to be a good leading indicator that any additional blip could cause 99 to hiccup.  I won't complain here though as 99% (ha, the irony) of use cases won't need it and the rare ones that do should still be able to get it from the raw data IIUC.

**Review by portante — COMMENTED** [review]
Ready for review, added support for 99.5 percentile handling.

**Review by dbutenhof — APPROVED** [review]

**Comment by portante** [comment]
@frival, we'll merge this once you sign off that this change provides what you are looking for, and we get another reviewer for this change.

**Review by zulcss — APPROVED** [review]
lgtm

**Review by dbutenhof — APPROVED** [review]

**Review by dbutenhof — APPROVED** [review]

**Review by npalaska — APPROVED** [review]

**Review by riya-17 — APPROVED** [review]

File: agent/bench-scripts/pbench-fio
Comment (ndokos) [inline agent/bench-scripts/pbench-fio:688]: Why are we using the full path to `mv` here and below? Guessing unit-test related: mocking `mv` does not affect these?

**Review by ndokos — APPROVED** [review]

**Review by ndokos — APPROVED** [review]

File: agent/bench-scripts/pbench-fio
Comment (portante) [reply agent/bench-scripts/pbench-fio:688]: I am guessing, but not sure entirely.  I tried not to change too much here.

**Pull request merged**
